src/search.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    cmp,
+    path::Path,
+    sync::mpsc::{self, Sender},
+};
+
+use crate::{utils, SearchBuilder};
+use ignore::{WalkBuilder, WalkState};
+use regex::Regex;
+
+/// A struct that holds the receiver for the search results
+///
+/// Can be iterated on to get the next element in the search results
+///
+/// # Examples
+///
+/// ## Iterate on the results
+///
+/// ```
+/// use rust_search::Search;
+///
+/// let search = Search::new("src", None, Some(".rs"), Some(1));
+///
+/// for path in search {
+///    println!("{:?}", path);
+/// }
+/// ```
+///
+/// ## Collect results into a vector
+///
+/// ```
+/// use rust_search::Search;
+///
+/// let search = Search::new("src", None, Some(".rs"), Some(1));
+///
+/// let paths_vec: Vec<String> = search.collect();
+/// ```
+pub struct Search {
+    rx: mpsc::Receiver<String>,
+}
+
+impl Iterator for Search {
+    type Item = String;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.rx.recv() {
+            Ok(v) => Some(v),
+            Err(_) => None,
+        }
+    }
+}
+
+impl Search {
+    /// Search for files in a given arguments
+    /// ### Arguments
+    /// * `search_location` - The location to search in
+    /// * `search_input` - The search input, defaults to any word
+    /// * `file_ext` - The file extension to search for, defaults to any file extension
+    /// * `depth` - The depth to search to, defaults to no limit
+    #[deprecated(
+        since = "1.0.0",
```

Comment:
Are we releasing this as a part of version 1.0.0 ?

Reply:
As i say above, we can release minor version because there is no Breaking change. So we can add more features to `SearchBuilder` before release `1.0.0`

Reply:
Alrighty..